src/User32.Desktop/User32.cs

```diff
@@ -621,10 +674,478 @@ public static partial class User32
+        /// </param>
+        /// <param name="lpsa">
+        /// A pointer to a <see cref="Kernel32.SECURITY_ATTRIBUTES"/> structure that determines whether the returned handle can be inherited by child processes. If lpsa is NULL, the handle cannot be inherited.
+        /// The <see cref="Kernel32.SECURITY_ATTRIBUTES.lpSecurityDescriptor"/> member of the structure specifies a security descriptor for the new desktop. If this parameter is NULL, the desktop inherits its security descriptor from the parent window station.
+        /// </param>
+        /// <param name="ulHeapSize">The size of the desktop heap, in kilobytes.</param>
+        /// <param name="pvoid">This parameter is also reserved and must be <see cref="IntPtr.Zero"/>.</param>
+        /// <returns>
+        /// If the function succeeds, the return value is a handle to the newly created desktop.
+        /// If the specified desktop already exists, the function succeeds and returns a handle to the existing desktop.
+        /// If the function fails, the return value is an invalid handle.
+        /// </returns>
+        /// <remarks>
+        /// <para>
+        /// If the dwDesiredAccess parameter specifies the READ_CONTROL, WRITE_DAC, or WRITE_OWNER standard access rights, you must also request the DESKTOP_READOBJECTS and DESKTOP_WRITEOBJECTS access rights.
+        /// </para>
+        /// <para>
+        /// The number of desktops that can be created is limited by the size of the system desktop heap, which is 48 MB. Desktop objects use the heap to store resources. You can increase the number of desktops that can be created by reducing the default heap reserved for each desktop in the interactive window station. This value is specified in the SharedSection substring of the following registry value: HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\SubSystems\Windows.
+        /// </para>
+        /// <para>
+        /// The default size of the desktop heap depends on factors such as hardware architecture. To retrieve the size of the desktop heap, call the <see cref="GetUserObjectInformation(IntPtr, ObjectInformationType, void*, uint, uint*)"/> function with UOI_HEAPSIZE.
+        /// </para>
+        /// </remarks>
+        [DllImport(nameof(User32), CharSet = CharSet.Unicode, ExactSpelling = true, SetLastError = true)]
+        public static unsafe extern SafeDesktopHandle CreateDesktopEx(
+           string lpszDesktop,
+           IntPtr lpszDevice,
+           IntPtr pDevmode,
+           DesktopCreationFlags dwFlags,
+           DESKTOP_ACCESS_MASK dwDesiredAccess,
+           [Friendly(FriendlyFlags.In | FriendlyFlags.Optional)] Kernel32.SECURITY_ATTRIBUTES* lpsa,
+           uint ulHeapSize,
+           IntPtr pvoid);
```

Comment:
This one is clever! I always tried to put optionals with IntPtr but it kept complaining it was not a constant and didn't let me do that, as well as string.Empty. Will take this for future reference.

Just used default with generics, don't know why, and never thought of it being used this way.

Way to go!!!! Perfect !!!!